### DISCUSSION TITLE
How to define Foreign Key in a value object

### DISCUSSION THREAD

**Discussion by liorbalmas** [abpframework/abp#7368] [Q&A] [answered]
Hi, 
I have this Value Object:
` 
public class Address : ValueObject
{
        public Guid CityId { get; private set; }
        public string Street { get; private set; }
        public int Number { get; private set; }
        public string Zipcode { get; private set; }
...
}
`

Address class has a CityId property pointing City class
Child class has an Address property.

Childs table has an Address_CityId column.
EF did not recognize it as FK of Cities.
When tried to set FK explicitly using this line:
`                b.HasOne<City>().WithMany().HasForeignKey(c => c.Address.CityId);
`
got this when updated migration:

`the expression 'c => Convert(c.Address.CityId, Object)' is not a valid member access expression. The expression should represent a simple property or field access: 't => t.MyProperty'. When specifying multiple properties or fields, use an anonymous type: 't => new { t.MyProperty, t.MyField }'. (Parameter 'memberAccessExpression')`

**Comment by amsaighi** [comment, marked as answer]
Hello, 
If Address is value object related to City Calss, you have to said that CityClass
`b.OwnsOne(x => x.Address , o =>
                {
                    o.Property(q => q.Street).HasColumnName(nameof(Address.Street));
                    o.Property(q => q.Number).HasColumnName(nameof(Address.Number));
                    o.Property(q => q.Zipcode).HasColumnName(nameof(Address.Zipcode));
                }); `
I removed cityId from the Address ( value object ) because they will added in the City.
If you wanna split your Class and Value object u have to add cityId in the value object.

And i did't catch this 

- Child class has an Address property.- 
- Childs table has an Address_CityId column.

Happy code

**Reply by liorbalmas** [reply]
Hi 
My mistake was having CityId in Address instead of a City.
Thanks
